### DISCUSSION TITLE
Differentiate between local and remote URLs

### DISCUSSION THREAD

**Discussion by dirkvranckaert** [OctoFarm/OctoFarm#1216] [Feature Request]
### OctoFarm Version

1.6.6

### OctoPrint Version(s)

1.8.3

### pm2 or docker?

docker

### Operating System

Mac OS

### Location of the problem

Web Interface

### Description of the problem

```markdown
So to configure the printers inside OctoFarm I configured them with the internal urls (using IP addresses in range 192.168.1.x off course). This seems way faster (as expected) than forcing OctoFarm to go around over the external access URLs of OctoPrint.
However pressing the web-button with a printer that also opens this internal URL, while often I'm not at the same location. So then I have to manually open the remote URL.
```

### Steps taken to reproduce

See description :-)

### What did you expect to happen

It would be nice to have the option to configure a 'secondary' URL that will be used (if defined) for launching your instance's OctoPrint web interface.

**Comment by NotExpectedYet** [comment]
Hey thanks for the report. 

I don't consider this a bug, more a feature request. 

I'd prefer personally to open a contained window with it wrapped but that requires turning on iframe on OP and me creating some messy routes. 

I'm not entirely sure having a remote and local IP is the best way to go about this but happy to discuss and flesh it out further. It would certainly increase the setup steps but I suppose that's more for people who would actually use the feature. Just feel there's a better way, dunno why aha. Will ponder on it more. 

Currently I take the standpoint that OP instances shouldn't be behind a reverse proxy so I have avoided supporting that use case. I am trying to build OctoFarm to be that interface through to them, so it can sit on the edge of a network type deal. 

I'll convert this to a discussion where my feature requests live anyway so please do pass your thoughts. Do you code JavaScript at all? I'd be happy with a PR but it's not setup at all to detect this currently aha.

**Comment by dirkvranckaert** [comment]
Well actually I would not open it inside octofarm in an iFrame or whatsover. Lots of people should not open up their OP instance to to be available publicly, that's what OP states itself since it cannot guarantee security. So only a minority will want to access the instance remotely with a proper setup (mostly security wise).
So for me it should not be a setup setup, it can be a power feature that you simply setup by editing the printers. You would then have the current URL and some 'secondary' URL. And only if that URL is set the button that now opens OP will then not use the default URL but instead the secondary.

I would not make it too complicated and not change the current setup flow since it's more a power feature to me.
Also this setup would be quite easy I think without intervening with or breaking any current features.

Let me know your thoughts!

**Comment by NotExpectedYet** [comment]
Yeah no to the iframe route, I was more thinking along the lines of tunnelling. 

I think rather than messing with routes/tunnels/urls it would probably be more advantageous to figure out why your going to the interface and remove the need to if you get me? 

The current implementation covers everything I ever bothered to access OctoPrint for personally, so for me it's feature complete. I know a couple of users have suggested that removing that need would be beneficial. The web interface button could just be hidden if it's not on a local url, and might be worth while doing regardless because then it's not readily clickable confusing users.
